### PR TITLE
fix: adds type error validations for `email` and `password` in login operation

### LINKED ISSUE #4822: Unhandled exceptions in login process
### Link to reproduction

_No response_

### Describe the Bug

The type of `email` field (and `password` too)[ is not checked in login operation](https://github.com/payloadcms/payload/blob/74c3fe1bb2e95e99b1416e72f9e78005624dd8a1/packages/payload/src/auth/operations/login.ts#L84). So if you send `POST` with empty body (or with empty email: `{ "email": "" }` to `/api/[any-auth-collection]/login` you will get `TypeError: Cannot read properties of null (reading 'toLowerCase')` in console and server will return 500 status code. Also if you send body where `email` field is not a string ( `{ "email": false }` | `{ "email": {} }` | `{ "email": 123 }` | etc) you will get another error (and 500 status code) in console: `TypeError: unsanitizedEmail.toLowerCase is not a function`

### To Reproduce

1. Create and run any payload app
2. Send empty POST request to any collection with auth (like `/api/users/login`)

### Payload Version

2.8.1

### Adapters and Plugins

_No response_

### PR DESCRIPTION
**Pull request by PatrikKozak**
## Description

Fixes #4822

Screenshot 2024-04-17 at 2 23 30 PM

Also updates error translations. There were missing translations for `error:localesNotSaved_one` & `error: localesNotSaved_other` in almost all translation files.

- [x] I have read and understand the [CONTRIBUTING.md](https://github.com/payloadcms/payload/blob/main/CONTRIBUTING.md) document in this repository.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] Existing test suite passes locally with my changes

### CODE REVIEW

File: packages/payload/src/auth/operations/login.ts [hunk 1]
```diff
@@ -83,6 +83,13 @@ async function login<TSlug extends keyof GeneratedTypes['collections']>(
 
     const { email: unsanitizedEmail, password } = data
 
+    if (typeof unsanitizedEmail !== 'string' || unsanitizedEmail.trim() === '') {
+      throw new Error('The following field is invalid: email')
```
Comment (DanRibbens) [inline packages/payload/src/auth/operations/login.ts:87]: This should be using translations from i18n. You can access it from `req`.

File: packages/payload/src/translations/ar.json [hunk 2]
```diff
@@ -69,6 +69,8 @@
     "invalidFileType": "نوع ملف غير صالح",
     "invalidFileTypeValue": "نوع ملف غير صالح: {{value}}",
     "loadingDocument": "حدثت مشكلة أثناء تحميل المستند برقم التعريف {{id}}.",
+    "localesNotSaved_one": "لم يتم حفظ اللغة التالية:",
```
Comment (DanRibbens) [inline packages/payload/src/translations/ar.json:72]: The `localesNotSaved` changes are from generating missing language and unrelated to this change, correct?

**Review by DanRibbens — CHANGES_REQUESTED** [review]
One little tweak to make.

File: packages/payload/src/translations/ar.json (hunk 2 above)
Comment (PatrikKozak) [reply packages/payload/src/translations/ar.json:72]: @DanRibbens Correct.

**Review by DanRibbens — APPROVED** [review]

**Pull request merged**